### DISCUSSION TITLE
flink cdc  sql  读取 oracle 全量问题

### DISCUSSION THREAD

**Discussion by cYfaris** [apache/flink-cdc#2014] [Q&A]
flink ：1.13
cdc：2.3
oracle ：12c

		String nocdb ="CREATE TABLE products (\n" +
				"     STU_NO STRING ,\n" +
				"     STU_NAME STRING , \n" +
				"     CARD_TYPE_CODE STRING \n" +
				"     ) WITH (\n" +
				"     'connector' = 'oracle-cdc',\n" +
				"     'hostname' = '172.16.67.88',\n" +
				"     'port' = '1521',\n" +
				"     'url' = 'jdbc:oracle:thin:@//172.16.67.88:1521/LHRSDB',\n" +
				"     'username' = 'flinkuser',\n" +
				"     'password' = 'flinkpw',\n" +
				"     'database-name' = 'LHRSDB',\n" +
				"     'schema-name' = 'FLINKUSER',\n" +
				"     'scan.startup.mode' = 'initial',\n" +
				"     'debezium.log.mining.strategy' = 'online_catalog',\n" +
				"     'debezium.log.mining.continuous.mine' = 'true',\n" +
				"     'debezium.database.tablename.case.insensitive'='false',\n"+
				"     'table-name' = 'NOCDB_STU_INFO')";

		tenv.executeSql(nocdb);
		TableResult tableResult = tenv.executeSql("select * from products");
		tableResult.print();

配置如上 读取全量是 只能读全量最后一条  且 无法续接增量   是哪里需要配置么
![image](https://user-images.githubusercontent.com/45968542/225861943-38786a27-cdea-4d66-90e8-72e14478ee1e.png)

**Comment by cYfaris** [comment]
解决了  主键的问题 对于sql  无主键情况下只能读一条

**Reply by lingyoucao** [reply]
怎么解决的？
